torchsynth/module.py

```diff
@@ -393,16 +376,14 @@ def _forward(self, midi_f0: T, mod_signal: Signal) -> Signal:
 
         """
         assert midi_f0.shape == (self.batch_size,)
-        assert (mod_signal >= -1).all() and (mod_signal <= 1).all()
 
         control_as_frequency = self.make_control_as_frequency(midi_f0, mod_signal)
         assert (control_as_frequency >= 0).all() and (
             control_as_frequency <= self.nyquist
         ).all()
 
         cosine_argument = self.make_argument(control_as_frequency)
-        cosine_argument += self.phase.unsqueeze(1)
-        self.phase.data = cosine_argument[:, -1]
+        cosine_argument += self.p("initial_phase").unsqueeze(1)
```

Comment:
Using initial phase here directly instead of the phase buffer. Because the phase buffer is initialized with the initial phase param during init, it doesn't get modified during randomization which means the phase of the oscillators is not deterministic. How do we feel about just using this here for now?